code/go/0chain.net/smartcontract/storagesc/allocation_pool.go

```diff
@@ -142,6 +144,47 @@ type allocationPool struct {
 	Blobbers          blobberPools     `json:"blobbers"`      //
 }
 
+func newAllocationPool(
+	t *transaction.Transaction,
+	alloc *StorageAllocation,
+	until common.Timestamp,
+	mintNewTokens bool,
+	balances chainState.StateContextI,
+) (*allocationPool, error) {
+	var err error
+	if !mintNewTokens {
+		if err = checkFill(t, balances); err != nil {
+			return nil, err
+		}
+	}
+
+	var ap allocationPool
+	var transfer *state.Transfer
+	if transfer, _, err = ap.DigPool(t.Hash, t); err != nil {
+		return nil, fmt.Errorf("digging write pool: %v", err)
+	}
+	if mintNewTokens {
+		balances.AddMint(&state.Mint{
```

Comment:
should we check the result of `balances.AddMint` ?

Reply:
tks